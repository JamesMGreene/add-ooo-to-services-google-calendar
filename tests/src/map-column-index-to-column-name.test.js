const mapColumnIndexToColumnName = require('../../src/map-column-index-to-column-name')

const AToZArray = Array(26)
const maximumColumnIndex = 27 * 26 - 1

describe('map-column-index-to-column-name', () => {
  it('returns expected values for inputs between 0 and 25', () => {
    const expectedOutput = AToZArray.map((e, i) => String.fromCharCode(65 + i))
    const actualOutput = AToZArray.map((e, i) => mapColumnIndexToColumnName(i))
    expect(actualOutput).toEqual(expectedOutput)
  })

  it('returns expected values for inputs between 26 and 51', () => {
    const expectedOutput = AToZArray.map(
      (e, i) => 'A' + String.fromCharCode(65 + i)
    )
    const actualOutput = AToZArray.map((e, i) =>
      mapColumnIndexToColumnName(26 + i)
    )
    expect(actualOutput).toEqual(expectedOutput)
  })

  it('returns expected values for inputs between 52 and 77', () => {
    const expectedOutput = AToZArray.map(
      (e, i) => 'B' + String.fromCharCode(65 + i)
    )
    const actualOutput = AToZArray.map((e, i) =>
      mapColumnIndexToColumnName(52 + i)
    )
    expect(actualOutput).toEqual(expectedOutput)
  })

  it('returns expected value for the maximum allowed input', () => {
    expect(mapColumnIndexToColumnName(maximumColumnIndex)).toBe('ZZ')
  })

  it('throws for input over the maximum allowed', () => {
    expect(() =>
      mapColumnIndexToColumnName(maximumColumnIndex + 1)
    ).toThrowErrorMatchingSnapshot()
  })
})
