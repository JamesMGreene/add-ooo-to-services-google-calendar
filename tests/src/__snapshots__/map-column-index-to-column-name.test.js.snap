// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map-column-index-to-column-name throws for input over the maximum allowed 1`] = `"Unexpectedly high column index: 702 (maximum allowed: 701)"`;
